packages/components/src/Alert/src/Alert.tsx

```diff
@@ -178,17 +187,18 @@ function Alert(props:AlertProps, ref: ForwardedRef<HTMLDivElement>) {
                                 <Button
                                     onPress={() => chain(renderProps.close(), onSecondaryButtonClick?.())}
                                     variant="secondary"
-                                    isDisabled={secondaryButtonDisabled}
+                                    isDisabled={isPrimaryButtonLoading || secondaryButtonDisabled}
                                     autoFocus={autoFocusButton === "secondary"}
                                 >
                                     {secondaryButtonLabel}
                                 </Button>
                             }
                             <Button
                                 variant={variant === "confirmation" ? "primary" : "danger"}
+                                isLoading={isPrimaryButtonLoading}
                                 isDisabled={primaryButtonDisabled}
                                 autoFocus={autoFocusButton === "primary"}
-                                onPress={() => chain(renderProps.close(), onPrimaryButtonClick?.())}
+                                onPress={() => chain(renderProps.close(), handleOnPrimaryButtonClick())}
```

Comment:
i think close should come after, since we want to put the loading first, THEN close